### ISSUE TITLE
A question about BCE loss function?

### ISSUE THREAD

**Issue by YangZeqin** [weitianxin/MACR#1]
Your paper is very helpful to me as a beginner who is interested in causal recommendation! But when I read the code, I encountered a little confusion. May I ask why the BCE loss function in the code considers positive and negative samples? Like the following code, it seems to be inconsistent with the formula (6) given in the paper?

`self.mf_loss_ori = tf.reduce_mean(tf.negative(tf.log(tf.nn.sigmoid(pos_scores)+1e-10))+tf.negative(tf.log(1-tf.nn.sigmoid(neg_scores)+1e-10)))`

![image](https://user-images.githubusercontent.com/48830280/128453136-11476140-f689-4513-af39-411f9bad56ce.png)

**Comment by weitianxin** [comment]
Thanks for your comment. The loss is the average loss over all samples, here we omit the data size |D|. The 1e-10 is used to avoid the log error (because log0=NaN). Deleting that will also be okay in most times.
